grakn-core/src/main/java/ai/grakn/graph/admin/ConceptCache.java

```diff
@@ -20,12 +20,52 @@
 
 import ai.grakn.concept.ConceptId;
 
+import java.util.Map;
+import java.util.Set;
+
 /**
  * Admin interface for Caching concepts which can be used later.
  *
  * @author Filipe Teixeira
  */
+//TODO: Remove bloat from this interface
```

Comment:
I suppose the main refactor is to make a keyspace-specific cache, seeing as nearly all these methods take `keyspace` as the first argument.